pkgs/development/python-modules/pyverilog/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+, jinja2
+, ply
+, verilog
+, pytest
+, pytest-pythonpath
+}:
+
+buildPythonPackage rec {
+  pname = "pyverilog";
+  version = "1.3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1a74k8r21swmfwvgv4c014y6nbcyl229fspxw89ygsgb0j83xnar";
+  };
+
+  disabled = !isPy3k;
+
+  propagatedBuildInputs = [
+    jinja2
+    ply
+    verilog
+  ];
+
+  checkInputs = [
+    pytest
+    pytest-pythonpath
+  ];
+
+  checkPhase = ''
+    py.test pyverilog tests
+  '';
+
+  PYVERILOG_IVERILOG = "${verilog}/bin/iverilog";
```

Comment:
Where do we need this? Please export it in the pre phase.

Reply:
So this is something I'm a bit confused about. It is a runtime dependency for the package. Had it been a standalone binary application, I would probably consider `makeWrapper`. The issue is without an abspath, the library won't work as desired. The alternative would be to leave it up to any program which uses the library to set, but I think this is more self-contained. Wouldn't exporting an environment variable in one of the pre phases only have a scope local to the build itself?

Reply:
Sure, but setting it here doesn't do much better.

Really, if you want to bind a python _library_ to a binary, your options are limited. Usually people end up digging into the source and patching in a hard-coded path. Perhaps making it _overridable_ by the environment variable if desired.

Reply:
Thank you, @risicle. @jonringer had [similar advice](https://www.reddit.com/r/NixOS/comments/oxo1wx/including_nonpython_runtime_dependencies_for/h7pofmy/). Will go with the patch.